VoiceMemos/Controller/VoicesTableViewController.swift

```diff
@@ -310,10 +293,47 @@ class VoicesTableViewController: BaseTableViewController, UISearchBarDelegate {
         }
     }
     
-    override func tableView(tableView: UITableView, editingStyleForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCellEditingStyle {
-        return .Delete
+    override func tableView(tableView: UITableView, editActionsForRowAtIndexPath indexPath: NSIndexPath) -> [UITableViewRowAction]? {
+        
+        let deleteAction = UITableViewRowAction(style: .Normal, title: "Delete"){ action, index in
+            
+            dispatch_async(dispatch_get_main_queue()){
+                let voice = self.voiceForRowAtIndexPath(indexPath, WithTableView: tableView)
```

Comment:
The handler block of the share action is also use the `voice` property, so you could consider to put this line of code at the beginning of the delegate method.